lib/routes/account.js

```diff
@@ -946,10 +1013,33 @@ module.exports = function (
         }
 
         function createResponse() {
-          return {
-             email: sessionToken.email,
-             verified: sessionToken.emailVerified
-           }
+
+          var doSigninConfirmation = requestHelper.shouldEnableSigninConfirmation(sessionToken.uid.toString('hex'), config)
+
+          if (doSigninConfirmation) {
+            return db.sessionTokenWithVerificationStatus(sessionToken.id)
+              .then(function (token) {
+
+                var sessionVerified = token.tokenVerified
```

Comment:
I just remembered @philbooth and I talking about this early on - the semantics of this endpoint have changed slightly, which is causing [this test failure in the fxa-js-client](https://github.com/mozilla/fxa-js-client/pull/203#issuecomment-222099775). If the user signs in w/o requesting keys and then calls `/recovery_email/status`, the response contains `verified: false`. This is because sessions that do not request keys are "unverified" by default, so that they cannot be used to get at Sync data.

In real life, returning `verified: false` shouldn't be a problem because `/recovery_email/status` is called after another endpoint indicates a user must verify their email. There is not really much of a reason to call `/recovery_email/status` if the other endpoints indicated the user was verified. Of course, there are some huge assumptions there - I have no idea when Fx Desktop, Fennec, iOS or Fx OS call `/recovery_email/status`.

If there a way to maintain backwards compatibility here, where we return `verified: false` only for accounts that are unverified, or sessions that request keys that have not yet confirmed the signin? I don't know what kind of shenanigans that would require.